src/child/actions/window.js

```diff
@@ -71,15 +78,22 @@ export function resizeToCompact() {
     };
 }
 
-export function resizeToDefault() {
+export function resizeToPrevious(dimensions, isMaximised) {
     return dispatch => {
         dispatch(resizing());
-        const defaultWindowDimensions = configService.getDefaultWindowDimensions();
         fin.desktop.Window.getCurrent().resizeTo(
-            defaultWindowDimensions[0],
-            defaultWindowDimensions[1],
+            dimensions[0],
+            dimensions[1],
             'top-right',
-            () => dispatch(expand()),
+            () => {
+                dispatch(expand());
+                if (isMaximised) {
```

Comment:
Without the `resizeTo()` first, a subsequent restore from maximize will cause the window to still be max sized but just in a different position. Using `resizeTo()` before `maximize()` ensured the restore would work correctly. Is there a better way to do this?